### DISCUSSION TITLE
Future of cantaloupe project

### DISCUSSION THREAD

**Discussion by janhoy** [UCLALibrary/docker-cantaloupe#145] [General]
The upstream project https://github.com/cantaloupe-project/cantaloupe has unfortunately been put to rest, with only one active maintainer/committer and no plans for any new releases (https://github.com/cantaloupe-project/cantaloupe/discussions/613)

We rely on this docker image and the cantaloupe server for a cultural heritage service in Norway, and worry the project may die or break into pieces. I proposed in the thread above to try to onboard more maintainers to keep the project healthy. Have you at UCLA library been in contact with the illinois to try to resovle the situation?

**Comment by ksclarke** [comment]
We were just discussing this yesterday too. I'll be interested to see the upstream response to your question on that thread. We've been doing security updates of packages in the Docker image (in fact, I have one I need to do this morning). We've not been running security/dependency checks over Cantaloupe itself. I did do that once with Snyk (which is what we use) and did see some areas of concern. I created a ticket in our system to report those to the upstream, but haven't made further progress with that. I had been assuming they were probably aware of it and didn't view them as critical (maybe a bad assumption?)

I do think there is a difference between a "pause" in the project and "being put to rest". My reading of adolski's response is that they're still using it, and will continue to use it, and would consider doing critical security bug fixes (though he seems to imply that minor security bugs will not be addressed). I think he's just saying: don't expect any new features or non-critical fixes (for awhile?) Fwiw, we have a PR in the Cantaloupe queue that has just been sitting there for quite awhile -- so I think this has been the case for some time now.

We did create a mechanism in our build where patchfiles could be applied, since we didn't see a lot of movement upstream in getting our PR merged: https://github.com/UCLALibrary/docker-cantaloupe/blob/60c714579e4f3f8190cebb05f25c9ceb86f94130/.github/workflows/release.yml#L75 We, of course, were thinking of this as a mechanism for smaller changes (and wanted to be cautious about sweeping changes that would be difficult to maintain with ongoing upstream updates).

If the project is indeed not going to be updated for security issues, _I think_ that's something we'd start doing. After all, we did choose the project, in part, because it was written in Java and would be familiar to our devs. I would guess we're not interested in making new feature changes, etc., to it (i.e., forking it). I have a meeting with our head of software dev today and can discuss the issue with him, though (e.g., whether there is any institutional support for greater involvement in Cantaloupe's development).

**Comment by ksclarke** [comment]
To more directly answer your question, though... we've not had any communication with the upstream other than submitting PRs and issues into their queue.

**Comment by janhoy** [comment]
Thanks. 

I hope, since Illinois created a separate GitHub org for the project, perhaps they hoped to make it more than a pure in-house project. Any project that size would benefit from having more than just a few active committers.
